lib/net.js

```diff
@@ -520,14 +534,18 @@ Socket.prototype.setNoDelay = function(enable) {
 };
 
 
-Socket.prototype.setKeepAlive = function(setting, msecs) {
+Socket.prototype.setKeepAlive = function(enable, initialDelayMsecs) {
   if (!this._handle) {
-    this.once('connect', () => this.setKeepAlive(setting, msecs));
+    this.once('connect', () => this.setKeepAlive(enable, initialDelayMsecs));
```

Comment:
You can avoid setting up this event listener now and just set `this[kSetKeepAlive]`, same for `kSetNoDelay`.

Reply:
Done!